database-plugins/src/main/java/io/cdap/plugin/DBRecord.java

```diff
@@ -162,16 +170,39 @@ private void setField(ResultSet resultSet, StructuredRecord.Builder recordBuilde
     Object o = DBUtils.transformValue(sqlType, sqlPrecision, sqlScale, resultSet, originalName,
                                       outputFieldSchema);
     if (o instanceof Date) {
+      bytesRead += Long.BYTES;
       recordBuilder.setDate(field.getName(), ((Date) o).toLocalDate());
     } else if (o instanceof Time) {
+      bytesRead += Integer.BYTES;
       recordBuilder.setTime(field.getName(), ((Time) o).toLocalTime());
     } else if (o instanceof Timestamp) {
+      bytesRead += Long.BYTES;
       Instant instant = ((Timestamp) o).toInstant();
       recordBuilder.setTimestamp(field.getName(), instant.atZone(ZoneId.ofOffset("UTC", ZoneOffset.UTC)));
     } else if (o instanceof BigDecimal) {
       BigDecimal decimal = ((BigDecimal) o);
+      bytesRead += decimal.unscaledValue().bitLength() / Byte.SIZE + Integer.BYTES;
       recordBuilder.setDecimal(field.getName(), decimal);
     } else {
+      Schema schema = field.getSchema();
+      if (schema.isNullable()) {
+        schema = schema.getNonNullable();
+      }
+      if (schema.getType() == Schema.Type.INT) {
```

Comment:
nit: use a switch-case instead of else-ifs ?

Reply:
fixed